cgen: generics contamination

### Describe the bug

Wrong generated C code `main__decode_slice_T_Array_fixed_int_3` doesn't make sense and `_v_dump_expr_Array_fixed_int_3` is been called inside `main__decode_T_main__Slice`

```c
_result_main__Slice main__decode_slice_T_main__Slice(void) {
	_result_main__Slice _t1 = {0};
	builtin___result_ok(&(main__Slice[]) { ((main__Slice){.pointer = ((void*)0),.len = 0,}) }, (_result*)(&_t1), sizeof(main__Slice));
	return _t1;
}
_result_Array_fixed_int_3 main__decode_slice_T_Array_fixed_int_3(void) {
	_result_Array_fixed_int_3 _t2;
	builtin___result_ok(&(Array_fixed_int_3[]) { {0} }, (_result*)(&_t2), sizeof(Array_fixed_int_3));
	#line 7 "/var/local/lib/isolate/0/box/code.v"
	_result_Array_fixed_int_3 _t1 = {0};
	memcpy(_t1.data, _t2.data, sizeof(Array_fixed_int_3));
	return _t1;
}
_result_void main__decode_T_main__Slice(void) {
	_result_main__Slice _t1 = main__decode_slice_T_main__Slice();
	if (_t1.is_error) {
		_result_void _t2 = {0};
		_t2.is_error = true;
		_t2.err = _t1.err;
		return _t2;
	}
	
	 _v_dump_expr_Array_fixed_int_3(_S("/var/local/lib/isolate/0/box/code.v"), 12, _S("decode_slice()!"), (*(main__Slice*)_t1.data));
	return (_result_void){0};
}
_result_void main__decode_T_Array_fixed_int_3(void) {
	_result_Array_fixed_int_3 _t1 = main__decode_slice_T_Array_fixed_int_3();
	if (_t1.is_error) {
		_result_void _t2 = {0};
		_t2.is_error = true;
		_t2.err = _t1.err;
		return _t2;
	}
	
	 _v_dump_expr_Array_fixed_int_3(_S("/var/local/lib/isolate/0/box/code.v"), 12, _S("decode_slice()!"), (*(Array_fixed_int_3*)_t1.data));
	return (_result_void){0};
}
```

### Reproduction Steps

```v
struct Slice {
	pointer &u8 = unsafe { nil }
	len     int
}

pub fn decode_slice[T]() !T {
	return T{}
}

@[manualfree]
pub fn decode[T]() ! {
	dump(decode_slice[T]()!)
}

fn main() {
	decode[Slice]()!
	decode[[3]int]()!
	println('ran')
}

```

### Expected Behavior

print 'ran'

### Current Behavior

```sh
>> compilation failed:
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/tsession_707b5e543740_01KD3CC6GK0M6QR4TT30FWQWZW/a_test.01KD3CC7BVFC0N0QM7S9B4VV0Y.tmp.c:8476: error: cannot convert 'struct main__Slice' to 'int *'
=================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.12 07331bd

### Environment details (OS name and version, etc.)

Linux
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/dumpexpr.v
+++ b/vlib/v/gen/c/dumpexpr.v
@@ -30,6 +30,9 @@ fn (mut g Gen) dump_expr(node ast.DumpExpr) {
 				name = g.styp(g.unwrap_generic(node.expr.info.typ.clear_flags(.shared_f,
 					.result))).replace('*', '')
 			}
+		} else if node.expr is ast.CallExpr {
+			name = g.styp(g.unwrap_generic(expr_type.clear_flags(.shared_f, .result))).replace('*',
+				'')
 		}
 	}
 	// var.$(field.name)

--- a/vlib/v/tests/generics/generic_dump_test.v
+++ b/vlib/v/tests/generics/generic_dump_test.v
@@ -1,0 +1,18 @@
+struct Slice {
+	pointer &u8 = unsafe { nil }
+	len     int
+}
+
+pub fn decode_slice[T]() !T {
+	return T{}
+}
+
+@[manualfree]
+pub fn decode[T]() ! {
+	dump(decode_slice[T]()!)
+}
+
+fn test_main() {
+	decode[Slice]()!
+	decode[[3]int]()!
+}

